geode-core/src/main/java/org/apache/geode/management/internal/beans/ManagementListener.java

```diff
@@ -97,121 +103,134 @@ private boolean shouldProceed(ResourceEvent event) {
-        GatewayReceiver stoppededRecv = (GatewayReceiver) resource;
-        adapter.handleGatewayReceiverStop(stoppededRecv);
-        break;
-      case GATEWAYSENDER_CREATE:
-        GatewaySender sender = (GatewaySender) resource;
-        adapter.handleGatewaySenderCreation(sender);
-        break;
-      case GATEWAYSENDER_START:
-        GatewaySender startedSender = (GatewaySender) resource;
-        adapter.handleGatewaySenderStart(startedSender);
-        break;
-      case GATEWAYSENDER_STOP:
-        GatewaySender stoppedSender = (GatewaySender) resource;
-        adapter.handleGatewaySenderStop(stoppedSender);
-        break;
-      case GATEWAYSENDER_PAUSE:
-        GatewaySender pausedSender = (GatewaySender) resource;
-        adapter.handleGatewaySenderPaused(pausedSender);
-        break;
-      case GATEWAYSENDER_RESUME:
-        GatewaySender resumedSender = (GatewaySender) resource;
-        adapter.handleGatewaySenderResumed(resumedSender);
-        break;
-      case GATEWAYSENDER_REMOVE:
-        GatewaySender removedSender = (GatewaySender) resource;
-        adapter.handleGatewaySenderRemoved(removedSender);
-        break;
-      case LOCKSERVICE_CREATE:
-        DLockService createdLockService = (DLockService) resource;
-        adapter.handleLockServiceCreation(createdLockService);
-        break;
-      case LOCKSERVICE_REMOVE:
-        DLockService removedLockService = (DLockService) resource;
-        adapter.handleLockServiceRemoval(removedLockService);
-        break;
-      case MANAGER_CREATE:
-        adapter.handleManagerCreation();
-        break;
-      case MANAGER_START:
-        adapter.handleManagerStart();
-        break;
-      case MANAGER_STOP:
-        adapter.handleManagerStop();
-        break;
-      case ASYNCEVENTQUEUE_CREATE:
-        AsyncEventQueue queue = (AsyncEventQueue) resource;
-        adapter.handleAsyncEventQueueCreation(queue);
-        break;
-      case ASYNCEVENTQUEUE_REMOVE:
-        AsyncEventQueue removedQueue = (AsyncEventQueue) resource;
-        adapter.handleAsyncEventQueueRemoval(removedQueue);
-        break;
-      case SYSTEM_ALERT:
-        AlertDetails details = (AlertDetails) resource;
-        adapter.handleSystemNotification(details);
-        break;
-      case CACHE_SERVER_START:
-        CacheServer startedServer = (CacheServer) resource;
-        adapter.handleCacheServerStart(startedServer);
-        break;
-      case CACHE_SERVER_STOP:
-        CacheServer stoppedServer = (CacheServer) resource;
-        adapter.handleCacheServerStop(stoppedServer);
-        break;
-      case LOCATOR_START:
-        Locator loc = (Locator) resource;
-        adapter.handleLocatorStart(loc);
-        break;
-      case CACHE_SERVICE_CREATE:
-        CacheService service = (CacheService) resource;
-        adapter.handleCacheServiceCreation(service);
-        break;
-      default:
-        break;
+    try {
+      if (!shouldProceed(event)) {
```

Comment:
`try` should be below this block otherwise the `finally` block will try to release a lock which had never been acquired.

Reply:
@jdeppe-pivotal thanks!